app/scripts/migrations/120.2.ts

```diff
@@ -26,7 +26,12 @@ export async function migrate(
   return versionedData;
 }
 
-function transformState(state: Record<string, unknown>) {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function transformState(state: Record<string, any>) {
```

Comment:
Use of `any` is disallowed by our TypeScript guidelines. The `unknown` type helps ensure we validate persisted state before using it, preventing further errors from appearing in migrations due to unexpected state corruption. Lets keep the `unknown` type here:

```suggestion
function transformState(state: Record<string, unknown>) {
```

See here on why we should avoid `any`, and on alternatives: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-any

Reply:
I see that in this case, it's hiding a TypeScript error caused by is not knowing whether `state.SnapControlelr` is an object.

We could handle that case by reporting a Sentry error and returning state unmodified. Probably best not to clear it as we're doing with SelectedNetworkController, just in case it has some recoverable data despite being the wrong type (e.g. what if we stringified it accidentally, then we could error and fix with a migration a later hot fix), but it's your call if you'd prefer console.warn + state clear for that case instead.

Reply:
I see that there is the potential for unexpected corruption in one state to impact the other. This is not ideal, but maybe not a deal breaker because we don't expect ti to happen. It just complicates the cleanup effort if it does happen. We could avoid this complication by refactoring the migration to use two separate functions to mutate the state here, one per controller, so they act more independently. Or we could move them to separate migrations. Not a blocker but that would be an improvement I think.

Reply:
How would you prefer the typing of `delete state.SnapController.snapErrors` then?

Reply:
Resolving because the `any` type has been removed.

We're still not reporting or fixing corrupted `SnapController` state. Typically we would in migrations, but I'm not sure how much that matters, I am not inclined to block on that.